src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -481,6 +492,27 @@ internal static bool TryGetAnnotation(this IEdmModel model, IEdmElement annotata
             return val;
         }
 
+        /// <summary>
+        /// Gets full name for the schema element with the provided namespace and name
+        /// </summary>
+        /// <param name="elementNamespace">Namespace of the element</param>
+        /// <param name="elementName">The element name</param>
+        /// <returns>The full name of the element</returns>
+        internal static string GetFullNameForSchemaElement(string elementNamespace, string elementName)
+        {
+            if (elementName == null)

```

Comment:
how about String.IsNullOrEmpty(elementName) to take care of case of empty string?
Same for the elementNameSpace below.

Reply:
I tried that and bunch of test cases failed. I wouldn't want to introduce a breaking change even if it's seemingly the right thing to do.

Reply:
Are those test errors for test case only or would introduce breaking changes?

Reply:
+1 to Biao's question, I'm curious on what cases failed

Reply:
Gist of my changes if performance. So I am not comfortable making changes that lead to any functional changes. For the large part I am replicating the exact code that existed before and instead of calculating the full name every time, I am caching it on the object and reusing it over and again.


I do not like to mix and match perf and functional changes. I can open a new issue for you to look at this.